R/model.R

```diff
@@ -414,17 +557,24 @@ compile_method <- function(quiet = TRUE,
   # - the stan model was changed since last compilation
   if (!file.exists(exe)) {
     force_recompile <- TRUE
-  } else if (file.exists(self$stan_file())
-             && file.mtime(exe) < file.mtime(self$stan_file())) {
-    force_recompile <- TRUE
+    regex_string <- paste0('^(',basename(exe_name_without_hash),'_)[a-f0-9]{16}')
+    regex_string <- cmdstan_ext(regex_string)
+    regex_string <- gsub('(.exe)$','(.exe)',regex_string)
+    regex_string <- paste0(regex_string,'$')
+    existing_exes <- list.files(
+      path = dirname(exe_name_without_hash)
+      , pattern = regex_string
+      , full.names = T
+    )
+    file.remove(existing_exes)
   }
 
   if (!force_recompile) {
     message("Model executable is up to date!")
     private$cpp_options_ <- cpp_options
-    private$precompile_cpp_options_ <- NULL
-    private$precompile_stanc_options_ <- NULL
-    private$precompile_include_paths_ <- NULL
+    # private$precompile_cpp_options_ <- NULL
+    # private$precompile_stanc_options_ <- NULL
+    # private$precompile_include_paths_ <- NULL
```

Comment:
Are these not necessary anymore? If not then go ahead and just delete them instead of commenting them out.

Reply:
I posted a comment asking what they were even there for in the first place. It seems odd to delete them when the model is compiled. I commented them out as a sort of marker that their removal was yet to be fully decided on. I can't think of any breaking changes that their removal would cause, so I'll commit their proper removal.